tests/app/test_frontend_proxy.py

```diff
@@ -1,83 +0,0 @@
-# Copyright 2018 New Vector Ltd
-#
-# Licensed under the Apache License, Version 2.0 (the "License");
-# you may not use this file except in compliance with the License.
-# You may obtain a copy of the License at
-#
-#     http://www.apache.org/licenses/LICENSE-2.0
-#
-# Unless required by applicable law or agreed to in writing, software
-# distributed under the License is distributed on an "AS IS" BASIS,
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.
-
-from synapse.app.generic_worker import GenericWorkerServer
-
-from tests.server import make_request
-from tests.unittest import HomeserverTestCase
-
-
-class FrontendProxyTests(HomeserverTestCase):
```

Comment:
unless we're actually killing off the frontend proxy (which we totally should, but I don't think we're there yet), perhaps this should stick around as an empty shell?

Reply:
As in, just an empty `class FrontendProxyTests`? Can do, but I don't really see how it helps with anything?

Reply:
you're probably right. In my head, we would check that a frontend proxy is a thing that can be instantiated - but that's a thing we check in sytest anyway (iirc?). It would also be a place ready for us to add tests for the other things that frontend proxies do, but I'm not sure I rate the chances of that happening any time soon.

Reply:
It's also worth noting that a frontend proxy is just a standard generic worker nowadays. But yes, we should ideally have tests for the fake `KeyUploadServlet` (or do we even need it know we can move encryption/device APIs off master?)

Reply:
> do we even need it know we can move encryption/device APIs off master?

well, good question. maybe we should try not using it?